### PR TITLE
fixed login

### PR DESCRIPTION
**Pull request by Asuraking1n**
Now Login returns user type, user ID also.
image

### CODE REVIEW

File: testOne/api/views.py [hunk 1]
```diff
@@ -123,8 +123,11 @@ def addcoach(request):
     serializer = CoachSerializer(data=request.data)
     if serializer.is_valid():
         serializer.save()
-        newUser = User(username=serializer.data['name'],password = serializer.data['password'])
+        newUser = User(username=serializer.data['name'],email=serializer.data['email'],password = serializer.data['password'])
```
Comment (jitendraag) [inline testOne/api/views.py:126]: User.objects.create_user -- This is the correct way to create a new user. The way you are using doesn't salt the password

File: testOne/api/views.py [hunk 2]
```diff
@@ -303,16 +306,25 @@ def addSession(request):
...
+        elif userType.type == 'faculty':
+            userProfile = Faculty.objects.get(email = email)
     except BaseException as e:
         raise ValidationError({"400":f'{str(e)}'})
     if password == Account.password:
         token = Token.objects.get_or_create(user = Account)
     else:
         raise ValidationError({"message": "Incorrect Login credentials"})
-    return Response({'status':'200','username':Account.username,'token':str(token[0])})
+    return JsonResponse({'status':'200','username':Account.username,'token':str(token[0]),'email':userProfile.email,'usertype':userType.type,"id":userProfile.id})
```
Comment (jitendraag) [inline testOne/api/views.py:327]: If we are doing raw json responses then we should consider having status and data objects separate.

File: testOne/api/views.py [hunk 3]
```diff
@@ -303,16 +306,25 @@ def addSession(request):
 @permission_classes([AllowAny])
 def login_user(request):
     userName = request.data['username']
+    email = request.data['email']
     password = request.data['password']
+    
     try:
         Account = User.objects.get(username = userName)
+        userType = Profile.objects.get(email = email)
+        if userType.type == 'coach':
```
Comment (jitendraag) [inline testOne/api/views.py:315]: Let's discuss this once. I thought we went through this psuedo code once.

**Review by jitendraag — CHANGES_REQUESTED** [review]
This is not how users should be saved. Please use create_user.

**Comment by kaaroMQ** [comment]
test


This was generated using python-pygraphviz, based on our model definitions 

Ref: https://gist.github.com/perrygeo/5380196

**Comment by kaaroMQ** [comment]
> test
> 
> This was generated using python-pygraphviz, based on our model definitions
> 
> Ref: https://gist.github.com/perrygeo/5380196

Here we can clearly see the disconnect between the Django based User we created and a link to Coach/Learner/Faculty profile we create.

Correct me if I am wrong here, we have tried 2 approach we experimented with @Nishant:

### Profile as Intermediate Model between User and User Details (in one of the 3 tables)  
* we have a relationship defined from a `Profile` -> `User` (we see this in the diagram)
* Add a relationship for one of `Coach`, `Learner`, or `Faculty` to Profile (missing yet)

### Direct relationship from `Coach`/`Learner` or `Faculty`
* Seems simpler to model,
* however, would require us to perform null checks, to infer the type of a user, based on the User object we get from Django.


As of now,  we are simply querying again a relevant table, based on the email, to find details of user, which seems like a recipe for something bad.

**Pull request merged**